src/plugins/links/public/embeddable/links_embeddable_factory.ts

```diff
@@ -128,8 +128,6 @@ export class LinksFactoryDefinition
     initialInput: LinksInput,
     parent?: DashboardContainer
   ): Promise<LinksEditorFlyoutReturn> {
-    if (!parent) return { newInput: {} };
-
```

Comment:
Similar to https://github.com/elastic/kibana/pull/170810/files#r1393019545, since the Link embeddable can now be edited outside of the Dashboard app, there is no guarantee that it has a parent - removing this early return is fine, because we already treat the parent as potentially undefined in the rest of the code.